### DISCUSSION TITLE
API to check if a PR is approved by a reviewer

### DISCUSSION THREAD

**Discussion by kalyan-nidumolu** [community/community#24346] [Apps, API and Webhooks]
<p>Hi,</p>
<p>How do I check if a PR has been approved ?<br>
I am currently using Github Python API.<br>
Created a repo object and get a PR object by doing:</p>
<p>pr=repo.get_pull(1602)</p>
<blockquote>
<blockquote>
<blockquote>
<p>type(pr)<br>
&lt;class ‘github.PullRequest.PullRequest’&gt;</p>
</blockquote>
</blockquote>
<blockquote>
<blockquote></blockquote>
</blockquote>
</blockquote>
<p>I don’t see an ‘approved’ attribute. How do I check if a PR is approved?</p>
<p>Thanks,<br>
Kalyan</p>

**Comment by nethgato** [comment, marked as answer]
<p>Hey there <a class="mention" href="/kalyan-nidumolu">@kalyan-nidumolu</a> :wave:</p>
<p>Indeed, there is a <code>state</code> value returned from this endpoint:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#reviews" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="20" height="20">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#reviews" target="_blank" rel="noopener">Pulls - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Syntax is:</p>
<pre><code class="lang-auto">GET /repos/{owner}/{repo}/pulls/{pull_number}/reviews
</code></pre>
<p>…and note that in the default response, we have:</p>
<pre><code class="lang-auto">"state": "APPROVED",
</code></pre>
<p>Let us know how you go!</p>

**Comment by kalyan-nidumolu** [comment]
<aside class="quote group-github-employees" data-username="nethgato" data-post="2" data-topic="141279">
<div class="title">
<div class="quote-controls"></div>
 nethgato:</div>
<blockquote>
<p>e that in the default response, we have:</p>
</blockquote>
</aside>
<p>Hi Nethgato,</p>
<p>Thanks for the response. What do you mean by “default response” ?</p>
<p>when I look at:<br>
<a href="https://api.github.com/repos/" rel="noopener nofollow ugc">https://api.github.com/repos/</a>//pulls//reviews</p>
<p>I only see an Array of reviews. For for each review, I see a state. If there is no approval, the state is “COMMENTED”. If the review has an approval, its state is “APPROVED”.</p>
<p>So I used:</p>
<p>def is_pr_approved(pr):<br>
review_list = pr.get_reviews()<br>
for review in review_list:<br>
if review.state == “APPROVED”:<br>
return True<br>
return False</p>
<p>Instead of iterating reviews like this - I was hoping to see if the top level PR object has “approved” flag or something.</p>
<p>But yeah… the current solution also works works for me.</p>

**Comment by nethgato** [comment]
<p>Hey <a class="mention" href="/kalyan-nidumolu">@kalyan-nidumolu</a></p>
<p>The default response is simply what we display in our documentation that exposes the <code>state</code> attribute and it’s subsequent values.</p>
<p>For the code you’re using, I feel less comfortable commenting and suggesting any changes as solving for your use case is not explicitly something that GitHub staff can assist with.</p>
<p>I would say if you’re hoping for coding assistance, to post in our  <a href="https://github.community/c/software-development/47">Software Development</a> board.</p>
<p>But for assistance with our APIs themselves, that’s what I’m kicking around this board for =)</p>

**Comment by kalyan-nidumolu** [comment]
<p>Hi <a class="mention" href="/nethgato">@nethgato</a>,</p>
<p>The code that I wrote was just for an example - to indicate that I currently don’t see that attribute in the  response from <a href="https://api.github.com/repos/owner/repo/pulls/pr-num/reviews" rel="noopener nofollow ugc">https://api.github.com/repos/owner/repo/pulls/pr-num/reviews</a></p>
<p>The response is just a List of reviews. I don’t see any default attributes other than that.<br>
The code, I just gave to explain the logic that I am currently using.</p>
<p>Thanks,<br>
Kalyan</p>

**Comment by nethgato** [comment]
<p>Thanks for further explaining <a class="mention" href="/kalyan-nidumolu">@kalyan-nidumolu</a> :bow:</p>
<p>Do you happen to have an example PR in a public repository that you could share?</p>

**Reply by julienbonastre** [reply]
This is trivial to understand... There is currently no API response that seems to achieve obtaining whether all required reviews have been satisfied
